src/core/server/http/http_config.ts

```diff
@@ -25,10 +25,27 @@ import { SslConfig, sslSchema } from './ssl_config';
 
 const validBasePathRegex = /^\/.*[^\/]$/;
 const uuidRegexp = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
+const ipv4Regexp = /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
+const ipv6Regexp = /^(?:[A-F0-9]{1,4}:){7}[A-F0-9]{1,4}$/i;
 
 const match = (regex: RegExp, errorMsg: string) => (str: string) =>
   regex.test(str) ? undefined : errorMsg;
 
+/** @internal */
+export const requestIdConfigSchema = schema.object({
+  allowFromAnyIp: schema.boolean({ defaultValue: true }),
```

Comment:
It feels like we should default this to `false`. The `X-Opaque-Id` is commonly used for HTTP requests, but it could be being used for a different purpose.